.taskcluster.yml

```diff
@@ -1,33 +1,58 @@
-# The version is always required
-version: 0
-allowPullRequests: public
-# Top level metadata is always required
-metadata:
-  name: "monopacker CI"
-  description: "Runs CI tasks for monopacker"
-  owner: "{{ event.head.user.email }}" # the user who sent the pr/push e-mail will be inserted here
-  source: "{{ event.head.repo.url }}" # the repo where the pr came from will be inserted here
+version: 1
+policy:
+  pullRequests: public
 tasks:
-  - provisionerId: "{{ taskcluster.docker.provisionerId }}"
-    workerType: "{{ taskcluster.docker.workerType }}"
-    metadata:
-      name: "Build Vagrant Box"
-      description: "Builds a vagrant box using packer"
-      owner: "{{ event.head.user.email }}"
-      source: "{{ event.head.repo.url }}"
-    extra:
-      github:
-        # Events that will trigger this task
-        events:
-          - pull_request.opened
-          - pull_request.synchronize
-          - push
-    payload:
-      maxRunTime: 3600
-      image: ubuntu
-      command:
-        - /bin/bash
-        - "--login"
-        - "-cx"
-        - >-
-          true # TODO FIXME
+  - $if: '(tasks_for == "github-pull-request"  && event["action"] in ["opened", "reopened", "synchronize"])'
+    then:
+      taskId: { $eval: as_slugid("test") }
+      created: { $fromNow: "" }
+      deadline: { $fromNow: "2 hours" }
+      provisionerId: aws-provisioner-v1
+      workerType: github-worker
+      payload:
+        maxRunTime: 3600
+        image: taskcluster/monopacker:build
+        command:
+          - /bin/bash
+          - --login
+          - -cx
+          - |
+            CHECKOUT_DIR="/monopacker_checkout"
+            git clone "${event.pull_request.head.repo.clone_url}" "$CHECKOUT_DIR"
+            cd "$CHECKOUT_DIR"
+            git checkout "${event.pull_request.head.sha}"
+            pipenv install
+            pipenv run make validate
+            pipenv run make test
+      metadata:
+        name: monopacker tests
+        description: runs tests, validates default packer builders
+        owner: taskcluster-internal@mozilla.com
+        source: ${event.pull_request.head.repo.clone_url}
+  - $if: 'tasks_for == "github-push"'
+    then:
+      taskId: { $eval: as_slugid("test") }
+      created: { $fromNow: "" }
+      deadline: { $fromNow: "2 hours" }
+      provisionerId: aws-provisioner-v1
+      workerType: github-worker
```

Comment:
Should we run all monopacker tasks on the workerpool we created for it or just the tasks that build images themselves?

Reply:
I'd say just the tasks that build images, but either way works. We have to figure out what that workerpool looks like.